### PR TITLE
Adding Scripting Functionality to OPI Creation Capabilities

### PR DESCRIPTION
**Pull request by GaryMcMurtrieS**

### CODE REVIEW

File: examples/script/opi_with_script.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+display.add_child(text_box)
+
+# Creating the rectangle - this is what will be updated by the script
+rectangle = widgets.Rectangle(5, 35, 50, 50)
+display.add_child(rectangle)
+
+# Creating the script object by passing it the path to the script, indicating we want it embedded,
+# and the script name.
+# Note that the script name only shows in the OPI editor.
+rectangle_script = Script(script_path="examples/script/embedded_script.py",
```
Comment (zhangt58) [inline examples/script/opi_with_script.py]: The value of `script_path` is not valid

File: opigen/renderers/phoebus/scripts.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+                    # Read the script file and embed it into the XML
+                    with open(script.script_path, 'r', encoding="utf-8") as file:
+                        script_text = file.read()
+
+                    et.SubElement(script_node, "text").text = et.CDATA(script_text)
+
+                # Non-embedded script
+                else:
+                    script_node = et.SubElement(scripts_node, "script", file=script.script_path)
```
Comment (zhangt58) [inline opigen/renderers/phoebus/scripts.py:28]: Here, we have the chance to strictly convert script path to absolute file path.

File: opigen/opimodel/scripts.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        """
+        self.script_path = script_path
+        self.embed = embed
+
+        if name is not None:
+            self.name = name
+        elif name is None:
+            # Formats the script path such that the name is only the file name without its file
+            # extension.
+            self.name = self.script_path.rsplit('/', 1)[-1].rsplit('.', 1)[0]
```
Comment (zhangt58) [inline opigen/opimodel/scripts.py]:
```
            self.name = self.script_path.rsplit('/', 1)[-1].rsplit('.', 1)[0]
```
Use `os.path.basename()`?

File: opigen/renderers/phoebus/scripts.py (hunk 2 above)
Comment (GaryMcMurtrieS) [reply opigen/renderers/phoebus/scripts.py:28]: What do you mean by that?

File: opigen/opimodel/scripts.py (hunk 3 above)
Comment (GaryMcMurtrieS) [reply opigen/opimodel/scripts.py]: We could, I was trying to avoid an unnecessary import

Comment (zhangt58) [reply opigen/opimodel/scripts.py]: No concerns for builtin modules.

Comment (GaryMcMurtrieS) [reply opigen/opimodel/scripts.py]: Fixed!

File: opigen/renderers/phoebus/scripts.py (hunk 2 above)
Comment (zhangt58) [reply opigen/renderers/phoebus/scripts.py:28]: This was not a good idea, I closed it.

**Pull request merged**